tsfresh/feature_extraction/feature_calculators.py

```diff
@@ -444,12 +444,17 @@ def partial_autocorrelation(x, param):
     if n <= 1:
         pacf_coeffs = [np.nan] * (max_demanded_lag + 1)
     else:
-        if (n <= max_demanded_lag):
-            max_lag = n - 1
+        # https://github.com/statsmodels/statsmodels/pull/6846
+        # PACF limits lag length to 50% of sample size.
+        if max_demanded_lag >= n // 2:
+            max_lag = n // 2 - 1
         else:
             max_lag = max_demanded_lag
-        pacf_coeffs = list(pacf(x, method="ld", nlags=max_lag))
-        pacf_coeffs = pacf_coeffs + [np.nan] * max(0, (max_demanded_lag - max_lag))
+        if max_lag > 0:
+            pacf_coeffs = list(pacf(x, method="ld", nlags=max_lag))
+            pacf_coeffs = pacf_coeffs + [np.nan] * max(0, (max_demanded_lag - max_lag))
+        else:
+            pacf_coeffs = [np.nan] * (max_demanded_lag + 1)
```

Comment:
Maybe I miss something here: but why the ` + 1`?

Reply:
the `lag` in param is from 0 to `max_demanded_lag`, so the length of `pacf_coeffs` should be `+1`, I found it when run test cases.

Reply:
Right! Sorry, that makes sense.